pandas/tests/reductions/test_stat_reductions.py

```diff
@@ -11,9 +11,74 @@
 
 import pandas as pd
 from pandas import DataFrame, Series, compat
+from pandas.core.arrays import DatetimeArray, PeriodArray, TimedeltaArray
 import pandas.util.testing as tm
 
 
+class TestDatetimeLikeStatReductions(object):
+    @pytest.mark.parametrize('box', [Series, pd.Index, DatetimeArray])
+    @pytest.mark.parametrize('tz', [None, 'US/Mountain'])
+    def test_dt64_mean(self, tz, box):
+        dti = pd.date_range('2001-01-01', periods=11, tz=tz)
+        # shuffle so that we are not just working with monotone-increasing
+        dti = dti.take([4, 1, 3, 10, 9, 7, 8, 5, 0, 2, 6])
+        dtarr = dti._data
+
+        obj = box(dtarr)
+        assert obj.mean() == pd.Timestamp('2001-01-06', tz=tz)
+        assert obj.mean(skipna=False) == pd.Timestamp('2001-01-06', tz=tz)
+
+        # dtarr[-2] will be the first date 2001-01-1
+        dtarr[-2] = pd.NaT
+
+        obj = box(dtarr)
+        assert obj.mean() == pd.Timestamp('2001-01-06 07:12:00', tz=tz)
+        assert obj.mean(skipna=False) is pd.NaT
+
+    @pytest.mark.parametrize('box', [Series, pd.Index, PeriodArray])
+    def test_period_mean(self, box):
+        dti = pd.date_range('2001-01-01', periods=11)
+        # shuffle so that we are not just working with monotone-increasing
+        dti = dti.take([4, 1, 3, 10, 9, 7, 8, 5, 0, 2, 6])
+
+        # use hourly frequency to avoid rounding errors in expected results
```

Comment:
not sure mean actually should work on PeriodIndex/Array

Reply:
how come?  the meaning is unambiguous

Reply:
I guess these have to be uniform periods so mean is just another period, though I think this could easily be a non-sensical return values, e.g. what is the mean of 

```
In [2]: pd.date_range('2012', freq='M', periods=4)                                                                                                                                                                                                                      
Out[2]: DatetimeIndex(['2012-01-31', '2012-02-29', '2012-03-31', '2012-04-30'], dtype='datetime64[ns]', freq='M')

```

Reply:
>  the meaning is unambiguous

Can you explain that more? Maybe from an implementation point (viewing it as integers) it is clear what to do, but semantically for periods, I certainly don't find this obvious.